code/lib/core-server/src/utils/new-story-templates/typescript.test.ts

```diff
@@ -0,0 +1,56 @@
+import { describe, expect, it } from 'vitest';
+import { getTypeScriptTemplateForNewStoryFile } from './typescript';
+
+describe('typescript', () => {
+  it('should return a TypeScript template with a default import', () => {
+    const result = getTypeScriptTemplateForNewStoryFile({
+      basename: 'foo',
+      componentExportName: 'default',
+      default: true,
+      frameworkPackageName: '@storybook/nextjs',
+      exportedStoryName: 'Default',
+    });
+
+    expect(result).toMatchInlineSnapshot(`
+      "import type { Meta, StoryObj } from '@storybook/nextjs';
+
+      import Component from './foo';
+
+      const meta = {
+        component: Component
+      } satisfies Meta<typeof Component>
+
+      export default meta;
+
+      type Story = StoryObj<typeof meta>;
```

Comment:
This is probably just me, but I actually prefer to define `type Story` _above_ the `const meta`, below the imports. In type-land it's legal to do a "forward reference" like that and I generally like to have all type declarations up top because it clarifies/avoids the implicit "hoisting" of types. (React component prop types are an exception, I always define those right before the component, to keep them together.)

Reply:
This is the way of how story files are documented in the docs. Also, personally, I tend to define the Story type after the meta definition, because reading the code like this is easier to understand, since you don't have to jump back and forth in the code to understand where `meta` is coming from.